cannot strip expression in this position - outer attributes not allowed

Not really sure what is the meaning of the message for : 
```
fn foo (e: &str) -> &str {
    &""
}
```
```
slice1.rs:2:6: error: cannot strip expression in this position - outer attributes not allowed
```

FTR, rustc gives : 
```
warning: unused variable: `e`
 --> .../git/gccrs/gcc/testsuite/rust.test/unsupported/slice1.rs:1:9
  |
1 | fn foo (e: &str) -> &str {
  |         ^ help: if this is intentional, prefix it with an underscore: `_e`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: function is never used: `foo`
 --> .../git/gccrs/gcc/testsuite/rust.test/unsupported/slice1.rs:1:4
  |
1 | fn foo (e: &str) -> &str {
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: 2 warnings emitted
```

--- a/gcc/rust/ast/rust-ast.h
+++ b/gcc/rust/ast/rust-ast.h
@@ -249,7 +249,8 @@ public:
     BYTE_STRING,
     INT,
     FLOAT,
-    BOOL
+    BOOL,
+    ERROR
   };
 
 private:
@@ -274,11 +275,11 @@ public:
 
   static Literal create_error ()
   {
-    return Literal ("", CHAR, PrimitiveCoreType::CORETYPE_UNKNOWN);
+    return Literal ("", ERROR, PrimitiveCoreType::CORETYPE_UNKNOWN);
   }
 
   // Returns whether literal is in an invalid state.
-  bool is_error () const { return value_as_string == ""; }
+  bool is_error () const { return type == ERROR; }
 };
 
 /* Forward decl - definition moved to rust-expr.h as it requires LiteralExpr to

--- a/gcc/rust/hir/rust-ast-lower-expr.h
+++ b/gcc/rust/hir/rust-ast-lower-expr.h
@@ -359,6 +359,10 @@ public:
       case AST::Literal::LitType::BOOL:
 	type = HIR::Literal::LitType::BOOL;
 	break;
+	// Error literals should have been stripped during expansion
+      case AST::Literal::LitType::ERROR:
+	gcc_unreachable ();
+	break;
       }
     auto crate_num = mappings->get_current_crate ();
     Analysis::NodeMapping mapping (crate_num, expr.get_node_id (),

--- a/gcc/testsuite/rust/compile/xfail/slice1.rs
+++ b/gcc/testsuite/rust/compile/xfail/slice1.rs
@@ -1,3 +1,5 @@
-fn foo (e: &str) -> &str {
-    &"" // { dg-bogus "cannot strip expression in this position - outer attributes not allowed" "#391" { xfail *-*-* } }
+// { dg-additional-options "-w" }
+
+fn foo(e: &str) -> &str { // { dg-bogus "expected" "#391" { xfail *-*-* } }
+    &"" // { dg-bogus "expected" "#391" { xfail *-*-* } }
 }

